ote/ote/modules/arg_parsers/image_classification.py

```diff
@@ -0,0 +1,69 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import argparse
+
+from ote.api import (train_args_parser,
+                     test_args_parser,
+                     export_args_parser,
+                     compression_args_parser)
+
+from .default import DefaultArgParser
+from ..registry import ARG_PARSERS
+
+
+@ARG_PARSERS.register_module()
+class ImageClassificationArgParser(DefaultArgParser):
+    # TODO(lbeynens): Rename to ReidArgParser
+
+    def get_train_parser(self, config_path):
+        parser = argparse.ArgumentParser(parents=[train_args_parser(config_path)], add_help=False)
+        parser = self._add_aux_weights_load_arg(parser)
+        parser = self._add_classes_arg(parser)
+        return parser
+
+    def get_test_parser(self, config_path):
+        parser = argparse.ArgumentParser(parents=[test_args_parser(config_path)], add_help=False)
+        parser = self._add_aux_weights_load_arg(parser)
+        parser = self._add_classes_arg(parser)
+        return parser
+
+    def get_export_parser(self, config_path):
+        parser = argparse.ArgumentParser(parents=[export_args_parser(config_path)], add_help=False)
+        return self._add_classes_arg(parser)
+
+    def get_compression_parser(self, config_path):
+        # Please, note that for image classification compression
+        # batch_size should be set from the compression config file,
+        # not from the command line arguments
+        # (smaller batch size often gives better results with "regularization by noise")
+        parser = argparse.ArgumentParser(parents=[compression_args_parser(config_path, with_batch_size=False)],
+                                         add_help=False)
+        parser = self._add_aux_weights_load_arg(parser)
+        parser = self._add_classes_arg(parser)
+        return parser
+
+    @staticmethod
+    def _add_classes_arg(parser):
+        parser.add_argument('--classes', required=False,
+                            help='Comma-separated list of classes (e.g. "cat,dog,mouse").')
+        return parser
+
+    @staticmethod
+    def _add_aux_weights_load_arg(parser):
+        parser.add_argument('--load-aux-weights', required=False,
```

Comment:
Without NNCF the initial weights for AUX models are ImageNet ones and downloaded by URLs inside deep-object-reid

Reply:
@LeonidBeynenson Can it be the same in case of NNCF?

Reply:
>@LeonidBeynenson Can it be the same in case of NNCF?
No. 
For NNCF the pair of checkpoints `(model_0.pth, model_1.pth)` should be taken as the result of the training process (from the same epoch).
That is because this option is required.

Reply:
Is it worth to set default values so that it automatically loads auxiliary model weights for training if they exist in working directory?

Reply:
>Is it worth to set default values so that it automatically loads auxiliary model weights for training if they exist in working directory?

I think not, since after the training the checkpoints for model_0 and model_1 will be in different subfolders. Moreover, after training there will be a lot of similar checkpoints in the same folder. I think the good way is to point the model for the auxiliary weight explicitly.

Please, note that I made a change in deep-object-reid submodule: now the framework automatically creates soft links to the best models (`model_0` and `model_1`) with the name `model-best.pth.tar` in the subfolders `model_0` and `model_1`:
* there is one soft link in the `model_0` subfolder and one soft link in the `model_1` subfolder
* these 2 soft links point to checkpoints with the same epoch
* the epoch is chosen as follows: it should have the best quality of `model_0` after the last LR drop

Reply:
@LeonidBeynenson I think it won't be clear for user how to select weights on its own.